eos-updater/poll.c

```diff
@@ -456,13 +474,22 @@ check_for_update_following_checkpoint_commits (OstreeRepo     *repo,
                               error))
     return FALSE;
 
-  out_update_ref_info->refspec = g_steal_pointer (&upgrade_refspec);
+  /* The "refspec" member is the *currently booted* refspec
+   * which may get cleaned up later if we change away from it. */
+  out_update_ref_info->refspec = g_steal_pointer (&booted_refspec);
+
+  /* The "remote", "ref" and "collection_ref" refer here to the
+   * ref and remote that we should be following given checkpoints. */
```

Comment:
I'm not super happy with this, but I basically took the approach of "it doesn't seem to be broken and I don't know if anything depends on this, so I'm not touching it". Ideally `ref` and `remote` would be the booted ref and remote.